### DISCUSSION TITLE
Need to get the Campaign Metrics Reports Immediately

### DISCUSSION THREAD

**Discussion by bipni** [amzn/ads-advanced-tools-docs#56] [Q&A] [answered]
Hi there,
I have been working in Report API to get the Metrics of a Campaign. From the documentation, I have learned that first, I need to create a report, after processing the operation I can download the report as a JSON file. It may take up to 3 hours to generate the report. But I need the report immediately after sending the request. Is there any way to achieve this?

Thanks in advance for your help.

**Comment by kdwivedi13** [comment, marked as answer]
Hey Bipni,

Thank you for initiating this thread; reporting API will undoubtedly take some time to prepare the requested report, however 3 hours is not always needed. But there certainly will be some waiting time. If you need an immediate response, you might want to explore through Amazon Marketing Stream, an event-driven architecture that provides reporting data.  Amazon Marketing Stream will deliver reporting data to your SQS Queue on an hourly basis. Please see the documentation for more information.
https://advertising.amazon.com/API/docs/en-us/amazon-marketing-stream/overview
https://github.com/amzn/amazon-marketing-stream-examples

I hope you find this information helpful.

**Reply by bipni** [reply]
Thanks a lot
